### PR TITLE
JSUI-2752 [DynamicFacet] Pass injectionDepth option

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/JSUI-2752

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: unitTests/controllers/DynamicFacetQueryControllerTest.ts [hunk 1]
```diff
@@ -58,6 +62,10 @@ export function DynamicFacetQueryControllerTest() {
       expect(facetRequest().type).toBe(facet.facetType);
     });
 
+    it('should send the injectionDepth', () => {
+      expect(facetRequest().injectionDepth).toBe(facet.options.injectionDepth);
```
Comment (olamothe) [inline unitTests/controllers/DynamicFacetQueryControllerTest.ts]: Should we not actually set the `injectionDepth` option in the test case for it to be meaningful ?

Comment (ThibodeauJF) [reply unitTests/controllers/DynamicFacetQueryControllerTest.ts]: updated

File: src/ui/DynamicFacet/DynamicFacet.ts [hunk 2]
```diff
@@ -227,7 +227,17 @@ export class DynamicFacet extends Component implements IDynamicFacet {
...
+
+    /**
+     * The *injection depth* to use.
+     *
+     * The injection depth determines how many results to scan in the index to ensure that the facet lists all potential
+     * facet values. Increasing this value enhances the accuracy of the listed values at the cost of performance.
+     *
+     * Default value is `1000`. Minimum value is `0`.
+     */
+    injectionDepth: ComponentOptions.buildNumberOption({ defaultValue: 1000, min: 0 })
```
Comment (fbeaudoincoveo) [inline src/ui/DynamicFacet/DynamicFacet.ts]: Oh! Seems like another dynamic facet limitation to cross off the list :)

File: src/ui/DynamicFacet/DynamicFacet.ts [hunk 3]
```diff
@@ -227,7 +227,17 @@ export class DynamicFacet extends Component implements IDynamicFacet {
...
-    dependsOn: ComponentOptions.buildStringOption()
+    dependsOn: ComponentOptions.buildStringOption(),
+
+    /**
+     * The *injection depth* to use.
+     *
+     * The injection depth determines how many results to scan in the index to ensure that the facet lists all potential
+     * facet values. Increasing this value enhances the accuracy of the listed values at the cost of performance.
+     *
+     * Default value is `1000`. Minimum value is `0`.
```
Comment (fbeaudoincoveo) [inline src/ui/DynamicFacet/DynamicFacet.ts]: Remove this line; will be autogenerated by TypeDoc

File: src/ui/DynamicFacet/DynamicFacet.ts [hunk 4]
```diff
@@ -227,7 +227,17 @@ export class DynamicFacet extends Component implements IDynamicFacet {
      *
      * @examples document-type-facet
      */
-    dependsOn: ComponentOptions.buildStringOption()
+    dependsOn: ComponentOptions.buildStringOption(),
+
+    /**
+     * The *injection depth* to use.
```
Comment (fbeaudoincoveo) [inline src/ui/DynamicFacet/DynamicFacet.ts]: Suggestion for the description:

> The number of items to scan for facet values.
>
> Setting this option to a higher value may enhance the accuracy of facet value counts at the cost of slower query performance.

**Pull request merged**